### ISSUE TITLE
Livestream.com

### ISSUE THREAD

**Issue by Junior1544** [chrippa/livestreamer#1422]
I'm trying to use livestreamer to play a stream from livestream.com and am having a very hard time...

Looking through the issues it looks like a few months ago there was some issues with this but haven't seen anything in a few months so I don't know if it's been fixed?

I'm a real newbie so any simple to follow instructions would be very helpful.  I can use omxplayer or vlc...

the stream I'm trying to play is:
https://livestream.com/Miraclenet/events/5004281

I'm just confused because I can't get it to go for more then like 3-4 seconds in omxplayer, and not at all in vlc...

Thanks!

**Comment by ghost** [comment]
This update (for Akamai HD streams) should work: #1277 

```
livestreamer "http://livestream.com/Miraclenet/events/5004281" best
[cli][info] Found matching plugin livestream for URL http://livestream.com/Miraclenet/events/5004281
[cli][info] Available streams: 486p_hls, 270p_hls, 432p_hls, 270p (worst), 432p, 486p (best)
[cli][info] Opening stream: 486p (akamaihd)
[cli][info] Starting player: "C:\Program Files (x86)\Video\MPC-HC\MPC-HC.exe"
```

It even works with the old plugin, if you switch to HLS.

```
livestreamer "http://livestream.com/Miraclenet/events/5004281" 486p_hls
[cli][info] Found matching plugin livestream for URL http://livestream.com/Miraclenet/events/5004281
[cli][info] Available streams: 486p_hls, 270p_hls, 432p_hls, 270p (worst), 432p, 486p (best)
[cli][info] Opening stream: 486p_hls (hls)
[cli][info] Starting player: "C:\Program Files (x86)\Video\MPC-HC\MPC-HC.exe"
```

**Comment by Junior1544** [comment]
that looks fine except it doesn't work on my Raspberry Pi 3 b....

**Comment by Junior1544** [comment]
oh, and I don't understand that other thread, but i tried the hls and that's what wasn't working right.  It'll play hls for a few seconds then sit's at a black screen and the only way to get the screen back is to reboot it..

**Comment by ghost** [comment]
I've tested it with VLC 2.1.1 and it works smoothly. There were some issues in the past with HLS streams and VLC, but this doesn't seem to be the case here.

If nobody has another idea, there is nothing we can do here to fix this problem.

**Comment by Junior1544** [comment]
Can someone help me to try the other solution 0880 wonderfully suggested?  I tried the hls one and it will play for about 5 seconds or so...  I wanted to try the other solution which seems to be to use a pull version but I have no idea how to do this with a raspberry pi (linux) since I'm just starting with this...   Will continue to research it myself as well but I hope someone can help :)

**Comment by Junior1544** [comment]
@scottbernstrin
Any thoughts on how i'd be able to implement that fix to livestreamer that was mentioned in that other thread and in #1277

**Comment by scottbernstein-zz** [comment]
Yes, just grab the complete livestreamer.py file mentioned in 1277 and replace the one that is in your plugins directory.

I have also tested this fix and it does indeed work for me.

**Comment by Junior1544** [comment]
This is to be run on a raspberry pi, and I can't seem to find the directory
for it at all or I would have done it.   Any help on finding the directory?
--James

**Comment by scottbernstein-zz** [comment]
Sorry I can't help you there -- I'm on a windows box.  Run a search on the pi for the file livestream.py

**Comment by Junior1544** [comment]
I did that and it doesn't show up. I'll keep searching.

--James

**Comment by scottbernstein-zz** [comment]
well if you can type livestreamer at the command prompt and it works, then it must be in your path.  Try typing "which livestreamer" at the command prompt and it should tell you where it's installed.

**Comment by intact** [comment]
@Junior1544 check your package manager

**Comment by Junior1544** [comment]
ok, I finally found it and made those changes to the livestream.py and it looks like it's working!  it still doesn't work for hls for more then a few seconds, but the regular one looks like it's working now...  I've run it for 5 minutes so far and it's running great!

Thanks for all the help guys!  My next task is to make it play the sound too because right now it play's the video part of the stream but there's no sound.

**Comment by stevekmcc** [comment]
For the record, the HLS issue was with omxplayer, which needs --timeout 20 otherwise it exits after 35-70 seconds (see streamlink issue [182](https://github.com/streamlink/streamlink/issues/182)). And probably the sound issue was that the Raspberry Pi sound was going to HDMI rather than analogue out. Your livestream command line should thus include: `--player 'omxplayer --timeout 20 -o local'`.

**Comment by Junior1544** [comment]
I have since moved to using the streamlink app...

Thanks guys and girls!

I'd suggest using the Streamlink app for anyone else because they have a team of people actually working on it and putting out releases and they're doing a great job!
